app_dart/test/request_handling/cache_request_handler_test.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+import 'package:mockito/mockito.dart';
+import 'package:neat_cache/neat_cache.dart';
+import 'package:test/test.dart';
+
+import 'package:cocoon_service/cocoon_service.dart';
+import 'package:cocoon_service/src/request_handling/body.dart';
+
+import '../src/datastore/fake_cocoon_config.dart';
+import '../src/request_handling/fake_http.dart';
+import '../src/request_handling/request_handler_tester.dart';
+
+void main() {
+  group('CacheRequestHandler', () {
+    FakeConfig config;
+    RequestHandlerTester tester;
+
+    Cache<Uint8List> cache;
+
+    const String testHttpPath = '/cache_request_handler_test';
+
+    setUp(() async {
+      config =
+          FakeConfig(redisResponseSubcacheValue: 'cache_request_handler_test');
+      tester = RequestHandlerTester(
+          request: FakeHttpRequest(
+        path: testHttpPath,
+      ));
+
+      final CacheService cacheService = CacheService(config);
+      cache = await cacheService.inMemoryCache(4);
+    });
+
+    test('returns response from cache', () async {
+      final RequestHandler<Body> fallbackHandlerMock = MockRequestHandler();
+
+      final Cache<Uint8List> responseCache =
+          cache.withPrefix(await config.redisResponseSubcache);
+
+      const String expectedResponse = 'Hello, World!';
+      final Body expectedBody = Body.forString(expectedResponse);
+
+      final Uint8List serializedBody =
+          await expectedBody.serialize().first;
+
+      await responseCache['$testHttpPath:'].set(serializedBody);
+
+      final CachedRequestHandler<Body> cacheRequestHandler =
+          CachedRequestHandler<Body>(
+              delegate: fallbackHandlerMock, cache: cache, config: config);
+
+      final Body body = await tester.get(cacheRequestHandler);
+      final Uint8List response = await body.serialize().first;
+      final String strResponse = String.fromCharCodes(response);
```

Comment:
We should use `utf8.decode()` here instead of `String.fromCharCodes()`

Reply:
Added. Is this because `String.fromCharCodes()` does UTF16?

Reply:
Yep.